packages/mui-material/src/BottomNavigationAction/BottomNavigationAction.test.js

```diff
@@ -73,4 +73,170 @@ describe('<BottomNavigationAction />', () => {
       expect(handleClick.callCount).to.equal(1);
     });
   });
+
+  describe('touch functionality', () => {
+    before(function test() {
+      // Only run in supported browsers
+      if (typeof Touch === 'undefined') {
+        this.skip();
+      }
+    });
+
+    let clock;
+
+    beforeEach(() => {
+      clock = useFakeTimers();
+    });
+
+    afterEach(() => {
+      clock.restore();
+    });
+
+    it('should fire onClick on touch tap', () => {
+      const handleClick = spy();
+
+      // Need disableTouchRipple to avoid ripple missing act (async setState after touchEnd)
+      const { container } = render(
+        <BottomNavigationAction onClick={handleClick} disableTouchRipple />,
+      );
+
+      fireEvent.touchStart(container.firstChild, {
+        touches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 42,
+            clientY: 42,
+          }),
+        ],
+      });
+
+      fireEvent.touchEnd(container.firstChild, {
+        changedTouches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 42,
+            clientY: 42,
+          }),
+        ],
+      });
+
+      clock.tick(100);
+
+      expect(handleClick.callCount).to.equal(1);
+    });
+
+    it('should not fire onClick twice on touch tap', () => {
+      const handleClick = spy();
+
+      // Need disableTouchRipple to avoid ripple missing act (async setState after touchEnd)
```

Comment:
You need to wrap the click and clock.tick in `act` anyway.

Reply:
Cool, i have removed disableRipple and wrapped clicks and clock.ticks in act. :+1: